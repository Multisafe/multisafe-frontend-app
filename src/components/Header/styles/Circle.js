import styled from "styled-components/macro";

export default styled.div`
  width: 48px;
  height: 48px;
  border-radius: 50%;
  padding: 13px 12px 11px;
  box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.16);
  background-color: ${({ theme }) => theme.primary};
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  font-size: 16px;

  svg:focus {
    outline: none;
  }
`;
