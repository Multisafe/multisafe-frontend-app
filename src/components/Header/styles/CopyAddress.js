import styled from "styled-components/macro";

export default styled.div`
  width: 48px;
  height: 48px;
  border-radius: 50%;
  margin: 0 0 0 8px;
  padding: 13px 12px 11px;
  box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.16);
  background-color: #7367f0;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;

  svg:focus {
    outline: none;
  }
`;
