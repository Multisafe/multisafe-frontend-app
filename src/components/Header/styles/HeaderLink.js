import { Link } from "react-router-dom";
import styled from "styled-components/macro";

export default styled(Link)`
  display: inline-flex;
  padding: 0.25em 0;
  margin: 1em 0;
  text-decoration: none;
  cursor: pointer;
  outline: 0;
  font-family: "Montserrat", sans-serif;
  font-weight: bold;
  font-size: 20px;
  color: ${({ theme }) => theme.logo.color};

  &:active {
    background: none;
    color: #fff;
  }

  &:hover {
    color: ${({ theme }) => theme.logo.color};
    text-decoration: none;
  }
`;
