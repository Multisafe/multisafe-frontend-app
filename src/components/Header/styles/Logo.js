import styled from "styled-components/macro";
import Logo from "assets/images/parcel-logo.png";

export default styled.div`
  background: url(${Logo}) no-repeat;
  background-size: cover;
  width: 60px;
  background-position: center;
`;
