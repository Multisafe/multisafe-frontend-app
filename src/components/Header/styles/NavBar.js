import styled from "styled-components/macro";

export default styled.div`
  min-height: 8rem;
  background-color: ${({ theme, white }) =>
    white ? "#f7f7f7" : theme.primary};

  box-shadow: ${({ white }) =>
    white ? "0 0.05rem 0.05rem 0 rgba(0, 0, 0, 0.15)" : "none"};
  color: #fff;
  border-bottom: solid 0.1rem rgba(255, 255, 255, 0.2);
`;
