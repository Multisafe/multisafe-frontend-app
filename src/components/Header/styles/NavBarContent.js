import styled from "styled-components/macro";

export default styled.div`
  max-width: 1200px;
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
