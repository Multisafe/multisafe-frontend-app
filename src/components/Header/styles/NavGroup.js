import styled from "styled-components/macro";

export default styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
