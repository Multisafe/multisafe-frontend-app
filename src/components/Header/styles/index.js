export { default as HeaderLink } from "./HeaderLink";
export { default as NavBar } from "./NavBar";
export { default as NavBarContent } from "./NavBarContent";
export { default as NavGroup } from "./NavGroup";
export { default as Logo } from "./Logo";
export { default as AccountAddress } from "./AccountAddress";
export { default as CopyAddress } from "./CopyAddress";
