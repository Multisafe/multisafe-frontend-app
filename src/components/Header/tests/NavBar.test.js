import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";
import { lightTheme } from "global-styles";

import { NavBar } from "../styles";

describe("<NavBar />", () => {
  it("should match snapshot", () => {
    const renderedComponent = renderer
      .create(
        <ThemeProvider theme={lightTheme}>
          <NavBar />
        </ThemeProvider>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("should have a class attribute", () => {
    const { container } = render(
      <ThemeProvider theme={lightTheme}>
        <NavBar />
      </ThemeProvider>
    );
    expect(container.querySelector("div").hasAttribute("class")).toBe(true);
  });
});
