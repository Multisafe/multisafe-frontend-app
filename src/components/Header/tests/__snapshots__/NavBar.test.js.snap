// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match snapshot 1`] = `
.c0 {
  border-bottom: 1px solid #eee;
  -webkit-transition: all 0.5s linear;
  transition: all 0.5s linear;
  height: 8vh;
}

<div
  className="c0"
/>
`;
